datalad/utils.py

```diff
@@ -79,20 +123,26 @@
 on_osx = platform_system == 'darwin'
 on_linux = platform_system == 'linux'
 
+T = TypeVar("T")
+K = TypeVar("K")
+V = TypeVar("V")
+P = ParamSpec("P")
```

Comment:
would it make sense to define some `datalad.typing` which would have above python version conditional imports and these variables defined so we could avoid needing to copy paste such things in every file needing to use similar constructs?

Reply:
Probably.

Reply:
Please do it then in this or subsequent PR.

Reply:
Done: b376ffca4